src/plugins/subscription-manager.py

```diff
@@ -159,7 +159,7 @@ def warnOrGiveUsageMessage(conduit):
             conduit.info(2, msg)
 
 
-def postconfig_hook(conduit):
+def posttrans_hook(conduit):
```

Comment:
It works, but the output is displayed in the middle of the installation which can be confusing:
```
[vagrant@centos7 vagrant]$ sudo yum install tree httpd
...
Running transaction
  Installing : httpd-2.4.6-88.el7.centos.x86_64                                                      1/2 
  Installing : tree-1.6.0-10.el7.x86_64                                                              2/2 
This system is registered with an entitlement server, but is not receiving updates. You can use subscription-manager to assign subscriptions.
  Verifying  : tree-1.6.0-10.el7.x86_64                                                              1/2 
  Verifying  : httpd-2.4.6-88.el7.centos.x86_64                                                      2/2 
...
```
It is probably cleaner to use `pretrans_hook`, which shows the display right after the `Running Transaction` message:
```
[vagrant@centos7 vagrant]$ sudo yum install tree httpd
...
Running transaction
This system is registered with an entitlement server, but is not receiving updates. You can use subscription-manager to assign subscriptions.
  Installing : apr-1.4.8-3.el7_4.1.x86_64                                                            1/6 
  Installing : apr-util-1.5.2-6.el7.x86_64                                                           2/6 
  Installing : httpd-tools-2.4.6-88.el7.centos.x86_64                                                3/6 
  Installing : mailcap-2.1.41-2.el7.noarch                                                           4/6 
  Installing : httpd-2.4.6-88.el7.centos.x86_64                                                      5/6 
  Installing : tree-1.6.0-10.el7.x86_64                                                              6/6 
  Verifying  : mailcap-2.1.41-2.el7.noarch                                                           1/6 
  Verifying  : httpd-tools-2.4.6-88.el7.centos.x86_64                                                2/6 
  Verifying  : apr-util-1.5.2-6.el7.x86_64                                                           3/6 
  Verifying  : tree-1.6.0-10.el7.x86_64                                                              4/6 
  Verifying  : httpd-2.4.6-88.el7.centos.x86_64                                                      5/6 
  Verifying  : apr-1.4.8-3.el7_4.1.x86_64                                                            6/6 
                                               1/1 
...
```

Reply:
I tested my change (and your proposal) on rhel and the change could not be so simple. Why? When you use RHEL, then all repositories are defined in redhat.repo. When the host is unregistered, then you cannot install any package and you will not see the message: `"This system is not registered with an entitlement server..."` I have some idea, but I would like to discuss it with the rest of the team.